### DISCUSSION TITLE
Add a Search function to the grid screen or as an options window.

### DISCUSSION THREAD

**Discussion by kf5ewy** [rrirower/DP2SceneGraphWizard#47] [Ideas]
Thank you for the great work on the wizard it has helped a lot with converting my channel over from direct publisher.  I  am new to  coding and was wondering how hard it would be to add a search function to be able to look up videos that are on the channel for people to find them more quickly. thank you

**Comment by rrirower** [comment]
Marked as a future enhancement.

**Reply by kf5ewy** [reply]
thank you. and thank you for the great wizard that you have created it has helped me a lot. keep up the good work

**Comment by kf5ewy** [comment]
Hello, I have been trying to work on this enhancement If i put my code here could you check and see if this would help with  the search function?  let me know if this  helps or hinders thank you.

`sub init()
    m.top.backgroundUri = "pkg/images/background.jpg"
    
      example = m.top.findNode("Example")

      examplerect = example.boundingRect()
      centerx = (1280 - examplerect.width) / 2
      centery = (720 - examplerect.height) / 2
      example.translation = [ centerx, centery ]
    
      m.top.setFocus(true)
    enterButton = m.top.findNode("enterButton")
    enterButton.observeField("buttonSelected", "onEnterButtonPressed")
    end sub

sub onEnterButtonPressed()
    'Get the text entered in the keyboard
    enteredText = m.top.findNode("keyboard").getEnteredText()

    sub searchChannelForContent()
        enteredText = m.top.dialog.getEnteredText() ' Get the text entered in the Keyboarddialog
        searchResults = searchChannel(enteredText) ' Call the function that performs the search

        ' Handle the search results...
    end sub

end sub

sub showdialog()
    keyboarddialog = createObject("roSGNode", "Keyboarddialog")
    keyboarddialog.backgroundUri = "pkg:images/keyboard.png"
    keyboarddialog.title = "Example"
    m.top.dialog = keyboarddialog
end sub

    function onKeyEvent(key as String, press as Boolean) as Boolean
      if press then
        if key = "OK"

          showdialog()

          return true
        end if
      end if

      return false
    end function`

**Reply by rrirower** [reply]
Thanks for the contribution. However, I'm knee deep in subscription code. This is low priority.

**Reply by kf5ewy** [reply]
i understand. Thank you for all the work you do to help us with roku channels.

**Comment by kf5ewy** [comment]
Hello @rrirower  I have been working on this search function and i have one that seems to work. still working on some bugs with it but it is functioning. I will share it with you if you would like to see if it will help.

**Reply by rrirower** [reply]
Thanks, but, I have one in the development pipeline.

**Comment by rrirower** [comment]
Available in beta3.
